### PR TITLE
DATAUP-557 `job_id_list` for `job_status` and `job_info` etc.

### PR DESCRIPTION
**Pull request by n1mus**
# Description of PR purpose/changes

* Refresh when getting `job.child_jobs` and `job.retry_ids` & rippling effects
* Add/update tests
* Refactors & bug fixes

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [n/a] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [n/a] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook
- [x] (Python) I have run Black and Flake8 on changed Python code manually or with a git precommit hook

# Updating Version and Release Notes (if applicable)

- [n/a] [Version has been bumped](https://semver.org/) for each release
- [n/a] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 1]
```diff
@@ -305,29 +310,22 @@ def lookup_job_state(self, job_id: str) -> dict:
...
+        self._verify_job_id_list(req)
+
         try:
-            job_state = self._jm.get_job_state(req.job_id)
-            self.send_comm_message("job_status", {req.job_id: job_state})
-            return job_state
-        except ValueError:
-            # kblogging.log_event(self._log, "lookup_job_state_error", {"err": str(e)})
+            output_states = self._jm.get_job_states(req.job_id_list)
+        except NoJobException:
```
[the remark this answers is not in the document]
Comment (ialarmedalien) [reply src/biokbase/narrative/jobs/jobcomm.py:322]: yes, possibly. We will take a look at that once the MVP is out, not now.

File: src/biokbase/narrative/jobs/jobcomm.py [hunk 2]
```diff
@@ -305,29 +310,22 @@ def lookup_job_state(self, job_id: str) -> dict:
...
             self.send_error_message("job_does_not_exist", req)
-            self.send_comm_message(
-                "job_status",
-                {
-                    req.job_id: {
-                        "state": {"job_id": req.job_id, "status": "does_not_exist"}
-                    }
-                },
-            )
```
[the remark this answers is not in the document]
Comment (ialarmedalien) [reply src/biokbase/narrative/jobs/jobcomm.py:327]: Hmmm, does unittest have something to check the array contents without looking at array order?

File: src/biokbase/narrative/tests/narrative_mock/mockclients.py [hunk 3]
```diff
@@ -503,29 +510,64 @@ def list(self):
...
 
-    def called(self, *args, **kwargs):
-        self.method_called = True
-        self.orig_method(*args, **kwargs)
+    aomc.assert_has_calls(
+        [
+            mock.call("fish", 1),
+            mock.call("dog", 2),
+        ]
+    )
```
Comment (ialarmedalien) [inline src/biokbase/narrative/tests/narrative_mock/mockclients.py:524]: 👍   Thanks for adding those docs!

File: src/biokbase/narrative/jobs/jobcomm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (n1mus) [reply src/biokbase/narrative/jobs/jobcomm.py:327]: Yeah, probably. It's the nested lists I sometimes have unexpected difficulties. I guess in this case `dict`s might be okay

File: src/biokbase/narrative/tests/narrative_mock/mockclients.py (hunk 3 above)
Comment (n1mus) [reply src/biokbase/narrative/tests/narrative_mock/mockclients.py:524]: Thank GitHub actions!

**Review by ialarmedalien — APPROVED** [review]
👍

**Review by briehl — APPROVED** [review]

**Comment by briehl** [comment]
I think this is ready to merge. Any last minute changes, @n1mus , or would you rather save those for the next steps?

**Comment by n1mus** [comment]
LGTM

**Pull request merged**